### ISSUE TITLE
Subitem not beeing saved

### ISSUE THREAD

**Issue by Cannonb4ll** [lukepolo/laracart#140]
I have the following code from your examples:

        $item = Cart::add(2, 'Shirt', 1, 15.99, [
            'size' => 'XXL'
        ]);

        $item->addSubItem([
            'description' => 'Extra Cloth Cost', // this line is not required!
            'price' => 3.00
        ]);

The problem is that the subitem 'Extra Cloth Cost' is not added to the item itself:

```
CartItem {#603 ▼
  #itemHash: "46c1b28fcfdaec4e8796943cc5a39a4e"
  #itemModel: null
  #itemModelRelations: []
  +locale: null
  +lineItem: false
  +discount: 0
  +subItems: []
  +couponInfo: []
  +internationalFormat: null
  +options: array:7 [▶]
}
```

What am I doing wrong?

**Comment by lukepolo** [comment]
It should work as you have it , whats in the options array?

**Comment by Cannonb4ll** [comment]
Yea I don't get it either, it just wont save, the complete cart model:

```
CartItem {#603 ▼
  #itemHash: "46c1b28fcfdaec4e8796943cc5a39a4e"
  #itemModel: null
  #itemModelRelations: []
  +locale: null
  +lineItem: false
  +discount: 0
  +subItems: []
  +couponInfo: []
  +internationalFormat: null
  +options: array:7 [▼
    "id" => 2
    "qty" => 3
    "name" => "Shirt"
    "taxable" => true
    "price" => 15.99
    "tax" => null
    "size" => "XXL"
  ]
}
```

**Comment by lukepolo** [comment]
My test looks like the code you have 

https://github.com/lukepolo/laracart/blob/master/tests/SubItemsTest.php#L45 

May have to look deeper into it .

**Comment by lukepolo** [comment]
```
are you doing 

  $item = Cart::add(2, 'Shirt', 1, 15.99, [
        'size' => 'XXL'
    ]);

    $item->addSubItem([
        'description' => 'Extra Cloth Cost', // this line is not required!
        'price' => 3.00
    ]);

dd($item)
```

**Comment by Cannonb4ll** [comment]
Yes but even without the dd() its not saving.

I am doing a redirect straight after, should this have any effect?

**Comment by lukepolo** [comment]
Well, it may be that the item isn't triggering the cart to save.

**Comment by lukepolo** [comment]
if you dont have the dd , add it to see if its even being added to the item. If it is , then its not firing the event to save the cart. If it isn't, then Ill have to load up a test site really quick and give it a test.

**Comment by Cannonb4ll** [comment]
If I dd() the subitem:

```
CartSubItem {#604 ▼
  +locale: null
  +internationalFormat: null
  -itemHash: "12cea28db004d8e9a5262f63cf2ac9e7"
  +options: array:3 [▼
    "items" => []
    "description" => "Extra Cloth Cost"
    "price" => 3.0
  ]
}
```

With this code:

```
        dd($item->addSubItem([
            'description' => 'Extra Cloth Cost', // this line is not required!
            'price' => 3.00
        ]));
```

**Comment by lukepolo** [comment]
do a dd on the item after doing addSubItem

**Comment by Cannonb4ll** [comment]
Then this is the result:

```
CartItem {#579 ▼
  #itemHash: "d249f704d2d702c5044bd5764ba14dc0"
  #itemModel: null
  #itemModelRelations: []
  +locale: null
  +lineItem: false
  +discount: 0
  +subItems: array:1 [▼
    "12cea28db004d8e9a5262f63cf2ac9e7" => CartSubItem {#604 ▼
      +locale: null
      +internationalFormat: null
      -itemHash: "12cea28db004d8e9a5262f63cf2ac9e7"
      +options: array:3 [▼
        "items" => []
        "description" => "Extra Cloth Cost"
        "price" => 3.0
      ]
    }
  ]
  +couponInfo: []
  +internationalFormat: null
  +options: array:7 [▼
    "id" => 2
    "qty" => 8
    "name" => "Shirt"
    "taxable" => true
    "price" => 15.99
    "tax" => null
    "size" => "XXL"
  ]
}
```

Code:

```
    public function add(Request $request, $id = null)
    {
        $item = Cart::add(2, 'Shirt', 1, 15.99, [
            'size' => 'XXL'
        ]);

        $item->addSubItem([
            'description' => 'Extra Cloth Cost', // this line is not required!
            'price' => 3.00
        ]);

        dd($item);

        return redirect()->action('CartController@show');
}
```

**Comment by lukepolo** [comment]
KK, it seems the cart isn't being saved after the addSubItem. cause the item has the sub item.

**Comment by lukepolo** [comment]
For now, you can call Cart::update(), ill have to make a fix on this!

**Comment by Cannonb4ll** [comment]
Thanks, this triggers the save I think, it gets added now.

**Comment by lukepolo** [comment]
Fixed, reference : 3e8338d1d66ffccfe2fcdfcfa17d5e72dea85ffc
